meshnet/jobs.go

```diff
@@ -30,3 +37,40 @@ func JobRefreshMeshnet(s *Server) func() error {
 		return nil
 	}
 }
+
+func JobMonitorFileshareProcess(s *Server) func() error {
+	wasFileshareUp := false
+	return func() error {
```

Comment:
It's a little bit more than `BlockFileshare` vs `AllowFileshare` because there is also `wasFileshareUp` state. And the `Block..` vs `Allow..` are called only on state change, so I can't just pull everything out of `if` blocks and determine only `Block..` vs `Allow..` in an if, but I can do something like this:
```
func JobMonitorFileshareProcess(s *Server) func() error {
	state := fileshareState{
		wasUp: false,
	}
	return func() error {
		state.isUp = internal.IsProcessRunning(internal.FileshareBinaryPath)

		var change func(UniqueAddress) error = nil
		switch state {
		case fileshareState{wasUp: true, isUp: false}:
			// fileshare process was running but it stopped - block fileshare ports
			change = s.netw.BlockFileshare
			state.wasUp = false
		case fileshareState{wasUp: false, isUp: true}:
			// fileshare process was down but now it is running - allow fileshare ports
			change = s.netw.AllowFileshare
			state.wasUp = true
		default:
			// No change is needed - either:
			// - fileshare was running and is running, or
			// - fileshare was down and is down
			// In both cases there is no state change.
		}

		if change != nil {
			peers, err := s.listPeers()
			if err != nil {
				return err
			}
			for _, peer := range peers {
				if err := change(UniqueAddress{UID: peer.PublicKey, Address: peer.Address}); err != nil {
					log.Println(internal.WarningPrefix, "failed to apply fileshare port change:", err)
				}
			}
		}

		return nil
	}
}
```
Is this better?

Reply:
Improved